### PR TITLE
#60 use package json

### PR DESCRIPTION
**Pull request by pYr0x**
package.json is now the main config file.
appName and sourceFolder will get from package.json

### CODE REVIEW

File: component/index.js [hunk 1]
```diff
@@ -63,27 +62,24 @@ module.exports = generators.Base.extend({
...
-				self.log.error("Main config file not found");
-				process.exit(1);
-			}
-			package = JSON.parse(fs.readFileSync(this.config.get('configMain'), 'utf8'));
-			folder = folder ? folder : _.get(package, 'system.directories.lib', folder);
-			appName = appName ? appName : _.get(package, 'name', appName);
-		}
+	  // try to read the packeage.json in the project root folder
+	  try {
+		  this.fs.accessSync(pkgFile, this.fs.R_OK);
```
Comment (matthewp) [inline component/index.js]: Why is it `this.fs.R_OK` and not just `fs.R_OK` like before?

Comment (pYr0x) [inline component/index.js]: i removed `var fs = require('fs');` since it is included in yeoman-generator

**Comment by matthewp** [comment]
Just one question, think @daffl should take a look since he knows more about yeoman than I do.

File: component/index.js (hunk 1 above)
Comment (matthewp) [inline component/index.js]: Ah, they probably add some more stuff to the module, makes sense.

**Comment by daffl** [comment]
Sweet. Looks good to me :shipit:

**Comment by matthewp** [comment]
This is a breaking change, so I'll release as 0.5.0

**Comment by matthewp** [comment]
@pYr0x I had to revert this because it broke the donejs/cli build: https://travis-ci.org/donejs/cli/jobs/105157029

Any ideas why that is?

**Comment by pYr0x** [comment]
i look into this.
the test said `No package.json file not found at /home/travis/build/donejs/cli/package.json` i have to run the test locally and figure out.
mybe we have to set a directory where a fresh package.json in placed for the test, like `donejs init` does

**Pull request merged**